src/main/java/fi/fmi/avi/archiver/config/MessagePopulatorFactoryConfig.java

```diff
@@ -1,37 +1,44 @@
 package fi.fmi.avi.archiver.config;
 
-import fi.fmi.avi.archiver.initializing.AviationProductsHolder;
+import fi.fmi.avi.archiver.config.model.AviationProduct;
 import fi.fmi.avi.archiver.message.populator.*;
 import fi.fmi.avi.model.GenericAviationWeatherMessage;
 import fi.fmi.avi.model.MessageType;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.core.convert.ConversionService;
 
-import javax.annotation.Resource;
 import java.time.Clock;
 import java.time.Duration;
 import java.util.Map;
 import java.util.Set;
 import java.util.regex.Pattern;
 
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
 @Configuration
 public class MessagePopulatorFactoryConfig {
-    @Autowired
-    private ConversionService conversionService;
-
-    @Autowired
-    private Clock clock;
 
-    @Autowired
-    private AviationProductsHolder aviationProductsHolder;
+    private final ConversionService conversionService;
+    private final Clock clock;
+    private final Map<String, AviationProduct> aviationProducts;
+    private final Map<MessageType, Integer> messageTypeIds;
+    private final Map<GenericAviationWeatherMessage.Format, Integer> messageFormatIds;
 
-    @Resource(name = "messageTypeIds")
-    private Map<MessageType, Integer> messageTypeIds;
+    public MessagePopulatorFactoryConfig(final ConversionService conversionService, final Clock clock, final Map<String, AviationProduct> aviationProducts,
```

Comment:
Though it's a bit more verbose, I'd consider using method parameter injection instead of constructor parameter injection + fields. This is because factory configurations may come and go - with method parameter injection it is explicit which dependencies are required by which bean. E.g. removing a bean from configuration also removes references to its dependencies.

Reply:
👍